packages/common/src/select/index.js

```diff
@@ -15,29 +20,143 @@ const handleSelectChange = (option, simpleValue, isMulti, onChange) => {
     : onChange(sanitizedOption);
 };
 
-class Select extends Component {
-  render() {
-    const { input, invalid, classNamePrefix, simpleValue, isMulti, pluckSingleValue, options, ...props } = this.props;
-    const { value, onChange, ...inputProps } = input;
+const selectProvider = (type) =>
+  ({
+    createable: CreatableSelect
+  }[type] || ReactSelect);
+
+const Select = ({
+  invalid,
+  classNamePrefix,
+  simpleValue,
+  isMulti,
+  pluckSingleValue,
+  options: propsOptions,
+  loadOptions,
+  loadingMessage,
+  loadingProps,
+  selectVariant,
+  updatingMessage,
+  noOptionsMessage,
+  value,
+  onChange,
+  loadOptionsChangeCounter,
+  ...props
+}) => {
+  const [state, dispatch] = useReducer(reducer, {
+    isLoading: false,
+    options: propsOptions,
+    promises: {},
+    isMounted: false
+  });
+
+  const updateOptions = () => {
+    dispatch({ type: 'startLoading' });
+
+    return loadOptions().then((data) => {
+      if (value && Array.isArray(value)) {
+        const selectValue = value.filter((value) =>
+          typeof value === 'object' ? data.find((option) => value.value === option.value) : data.find((option) => value === option.value)
+        );
+        onChange(selectValue.length === 0 ? undefined : selectValue);
+      } else if (value && !data.find(({ value: internalValue }) => internalValue === value)) {
+        onChange(undefined);
+      }
+
+      dispatch({ type: 'updateOptions', payload: data });
+    });
+  };
+
+  useEffect(() => {
+    if (loadOptions) {
+      updateOptions();
+    }
+
+    dispatch({ type: 'mounted' });
```

Comment:
This is not reliable, you should use `useRef` for cases like this. There is an example: https://gist.github.com/jaydenseric/a67cfb1b809b1b789daa17dfe6f83daa